components/src/dynamo/vllm/checkpoint/checkpointable_async_llm.py

```
@@ -0,0 +1,1166 @@
+        assert metadata.zmq_port is not None
+        self.port = metadata.zmq_port
+        self.socket_url = f"tcp://127.0.0.1:{self.port}"
+        logger.info("Using saved port %d from checkpoint", self.port)
+
+        # Ensure the original tree PID from dump is fully gone to avoid
+        # PID collisions when restoring into the same PID namespace.
+        if metadata.tree_pid:
+            start = time.time()
+            # Wait up to a short grace period since dump should have killed it
+            while time.time() - start < 5.0:
+                if os.system(f"kill -0 {metadata.tree_pid} >/dev/null 2>&1") != 0:
```

Ruff (0.14.3)

Starting a process with a shell, possible injection detected

(S605)